titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,245 @@
+import numpy
+from cogeo_mosaic.backends import MosaicBackend
+from cogeo_mosaic.mosaic import MosaicJSON
+from cogeo_mosaic.utils import get_footprints
+from rasterio.transform import from_bounds
+from rio_tiler.io.cogeo import tile as cogeoTiler
+from rio_tiler.profiles import img_profiles
+from rio_tiler.utils import render
+
+from titiler.api.deps import CommonTileParams, TileMatrixSetNames
+from titiler.api.endpoints.cog import cog_info, tile_response_codes
+from titiler.api.utils import postprocess
+from titiler.models.metadata import cogBounds, cogInfo
+from titiler.models.mosaic import CreateMosaicJSON, UpdateMosaicJSON
+from titiler.ressources.common import drivers
+from titiler.ressources.enums import ImageMimeTypes, ImageType, PixelSelectionMethod
+from titiler.ressources.responses import ImgResponse
+
+from fastapi import APIRouter, Depends, Path, Query
+from fastapi.routing import APIRoute
+
+from starlette.concurrency import run_in_threadpool
+from starlette.exceptions import HTTPException
+from starlette.requests import Request
+from starlette.responses import Response
+
+
+def _chunks(my_list: Sequence, chuck_size: int):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(my_list), chuck_size):
+        yield my_list[i : i + chuck_size]
+
+
+class MosaicJSONRouter(APIRoute):
+    """Custom router to temporarily forbid usage of cogeo-mosaic STAC backend"""
+
+    def get_route_handler(self) -> Callable:
+        """Override base method (https://fastapi.tiangolo.com/advanced/custom-request-and-route/)"""
+        original_route_handler = super().get_route_handler()
+
+        async def forbid_stac_backend(request: Request) -> Response:
+            url = request.query_params.get("url")
+            if not url:
+                body = await request.json()
+                url = body.get("url")
+            if url.startswith("stac"):
+                # Raise as a validation error
+                raise HTTPException(
+                    status_code=422,
+                    detail="STAC is not currently supported for mosaicjson endpoints",
+                )
+            return await original_route_handler(request)
+
+        return forbid_stac_backend
+
+
+router = APIRouter()
+
+
+@router.post(
+    "", response_model=MosaicJSON, response_model_exclude_none=True,
+)
+def create_mosaicjson(body: CreateMosaicJSON):
+    """Create a MosaicJSON"""
+    mosaic = MosaicJSON.from_urls(
+        body.files,
+        minzoom=body.minzoom,
+        maxzoom=body.maxzoom,
+        max_threads=body.max_threads,
+    )
+    with MosaicBackend(body.url, mosaic_def=mosaic) as mosaic:
+        mosaic.write()
+        return mosaic.mosaic_def
+
+
+@router.get(
+    "",
+    response_model=MosaicJSON,
+    response_model_exclude_none=True,
+    responses={200: {"description": "Return MosaicJSON definition"}},
+)
+def read_mosaicjson(
+    resp: Response, url: str = Query(..., description="MosaicJSON URL")
```

Comment:
if we follow the `user.layer` schema, what do you think about adding this in the path ? 

e.g `endpoint/mosaicjson/kylebarron.somethingawesome`

Reply:
If we went with `user.layer` I think is the most "restful" way (`/endpoint/resource/<id>`).

The more I think about that only EFS and S3 support write methods the more I like this idea.